src/halmos/__main__.py

```diff
@@ -819,19 +836,25 @@ def main() -> int:
     # compile
     #
 
-    try:
-        crytic_compile_args, crytic_compile_unknown_args  = crytic_compile_parser.parse_known_args()
+    build_cmd = [
+        'forge', # shutil.which('forge')
+        'build',
+        '--root', args.root,
+        '--extra-output', 'storageLayout', 'metadata'
+    ]
 
-        both_unknown = set(halmos_unknown_args) & set(crytic_compile_unknown_args)
-        if both_unknown:
-            print(color_warn(f'error: unrecognized arguments: {" ".join(both_unknown)}'))
-            return 1
+    # run forge without capturing stdout/stderr
+    build_exitcode = subprocess.run(build_cmd).returncode
 
-        cryticCompile = CryticCompile(target=args.root, **vars(crytic_compile_args))
-    except InvalidCompilation as e:
-        print(color_warn(f'Parse error: {e}'))
+    if build_exitcode:
+        print(color_warn(f'build failed: {build_cmd}'))
         return 1
 
+    try:
+        build_out = parse_build_out(args)
+    except Error as err:
+        print(color_warn(f'build output parsing failed: {err}'))
```

Comment:
I think there's a problem with the except block

```
    except Error as err:
NameError: name 'Error' is not defined. Did you mean: 'OSError'?
```

Reply:
```suggestion
    except Exception as ex:
        print(color_warn(f'build output parsing failed: {ex}'))
```

Reply:
should we continue after this exception or abort? 

if we continue, it fails with 

```
    for compiler_version in sorted(build_out):
UnboundLocalError: local variable 'build_out' referenced before assignment
```

Reply:
good catch, thanks!!